stock_measuring_device/models/measuring_device.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+import logging
+
+from odoo import _, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class MeasuringDevice(models.Model):
+    _name = "measuring.device"
+    _inherit = "collection.base"
+    _description = "Measuring and Weighing Device"
+    _order = "warehouse_id, name"
+
+    name = fields.Char("Name", required=True)
+    warehouse_id = fields.Many2one("stock.warehouse", "Warehouse", required=True)
+    device_type = fields.Selection(
+        selection=[],
+        help="The type of device (e.g. zippcube, cubiscan...) "
+        "depending on which module are installed.",
+    )
+    state = fields.Selection(
+        [("not_ready", "Not Ready"), ("ready", "Ready")],
+        default="not_ready",
+        readonly=True,
+        copy=False,
+    )
+
+    _sql_constraints = [
+        (
+            "name_uniq",
+            "unique (name)",
+            "The name of the measuring/weighing device must be unique.",
+        ),
+    ]
+
+    def _forward(self, method, *args, **kwargs):
+        _logger.debug(
+            "Measuring Device ID: %s type %s > %s : %s %s",
+            self.id,
+            self.device_type,
+            method,
+            args,
+            kwargs,
+        )
+        self.ensure_one()
+        measuring_device = self._get_measuring_device()
+        return getattr(measuring_device, method)(*args, **kwargs)
+
+    def _get_measuring_device(self):
+        with self.work_on(self._name) as _forward:
+            return _forward.component(usage=self.device_type)
+
```

Comment:
```suggestion
        with self.work_on(self._name) as work:
            return work.component(usage=self.device_type)

```

Reply:
you had requested me to use _forward in your previous review...

Reply:
no, I asked you to use call `_forward` to forward calls to the device, not to call `_forward` the `work` context of the component :wink: